intelmq/bots/outputs/templated_smtp/output.py

```diff
@@ -0,0 +1,187 @@
+        if not self.body:
+            raise ConfigurationError("Mail", "No body specified")
+
+        self.templates = {
+            "subject": Template(self.subject),
+            "from": Template(self.mail_from),
+            "to": Template(self.mail_to),
+            "body": Template(self.body),
+            "attachments": []
+        }
+        for att in self.attachments:
+            if "content-type" not in att:
+                self.logger.error("Attachment does not have a content-type, ignoring: %s.", att)
+                continue
+            elif "text" not in att:
+                self.logger.error("Attachment does not have a text, ignoring: %s.", att)
+                continue
+            elif "name" not in att:
+                self.logger.error("Attachment does not have a name, ignoring: %s.", att)
+                continue
+            attachment_template = {
+                "content-type": Template(att["content-type"]),
+                "text": Template(att["text"]),
+                "name": Template(att["name"])
+            }
+            self.templates["attachments"].append(attachment_template)
+
+    def process(self):
+        event = self.receive_message()
+
+        if self.verify_cert and self.smtp_class == smtplib.SMTP_SSL:
+            kwargs = {'context': ssl.create_default_context()}
+        else:
+            kwargs = {}
+
+        with self.smtp_class(self.smtp_host, self.smtp_port,
+                             **kwargs) as smtp:
+            if self.starttls:
+                if self.verify_cert:
+                    smtp.starttls(context=ssl.create_default_context())
+                else:
+                    smtp.starttls()
+            if self.username and self.password:
+                smtp.login(user=self.username, password=self.password)
+
+            msg = EmailMessage()
+            msg['Subject'] = self.templates["subject"].render(event=event)
+            msg['From'] = self.templates["from"].render(event=event)
+            msg['To'] = self.templates["to"].render(event=event)
+            msg.add_attachment(self.templates["body"].render(event=event), disposition='inline')
+
+            for att in self.templates["attachments"]:
+                msg.add_attachment(att["text"].render(event=event).encode('utf-8'),
+                                   maintype=att["content-type"].render(event=event).split('/')[0],
+                                   subtype=att["content-type"].render(event=event).split('/')[1],
+                                   disposition='attachment',
+                                   filename=att["name"].render(event=event))
+
+            recipients = [recipient.strip()
+                          for recipient
+                          in msg['To'].split(',')]
+            self.logger.info("Sending mail to %s.", recipients)
+            smtp.send_message(msg, from_addr=msg['From'],
+                              to_addrs=recipients)
+
```

Comment:
Can we add a try catch block around the `send_message` call and add some debug logging?

Reply:
I'm not sure what that would accomplish. Any errors from smtplib will already cause an exception to propagate upwards where the framework will log it, dump the event to disk so it isn't lost and restart the bot. I don't think the bot can do anything more intelligent by itself.